borg/remote.py

```diff
@@ -28,6 +29,21 @@
 MAX_INFLIGHT = 100
 
 
+def os_write(fd, data):
+    """os.write wrapper so we do not lose data for partial writes."""
+    # This is happening frequently on cygwin due to its small pipe buffer size of only 64kiB
+    # and also due to its different blocking pipe behaviour compared to Linux/*BSD.
+    # Neither Linux nor *BSD ever do partial writes on blocking pipes, unless interrupted by a
+    # signal, in which case serve() would terminate.
+    amount = len(data)
+    while data:
+        count = os.write(fd, data)
+        data = data[count:]
```

Comment:
btw, i linked to the ticket and this pr in emails to the cygwin mailing list. so, hopefully someone reads this and fixes any bugs in cygwin.